crates/wash-cli/src/dev.rs

```diff
@@ -1527,22 +1534,35 @@ async fn run_dev_loop(
         emoji::CONSTRUCTION_BARRIER,
         style("Building project...").bold(),
     );
-    let artifact_path = build_project(project_cfg, Some(&SignConfig::default()))
+    let Ok(Ok(artifact_path)) = build_project(project_cfg, Some(&SignConfig::default()))
         .await
-        .context("failed to build project")?
-        .canonicalize()
-        .context("failed to canonicalize path")?;
+        .context("failed to build project")
+        .map(|p| p.canonicalize().context("failed to canonicalize path"))
+    else {
+        eprintln!(
+            "{} {}",
+            emoji::ERROR,
+            style("Failed to build project").red()
+        );
+        // Failing to build the project can be corrected by changing the code and shouldn't
+        // stop the development loop
+        return Ok(());
+    };
     eprintln!(
-        "✅ successfully built project at [{}]",
+        "{} Successfully built project at [{}]",
+        emoji::GREEN_CHECK,
         artifact_path.display()
     );
     let component_ref = format!("file://{}", artifact_path.display());
 
     // After the project is built, we must ensure dependencies are set up and running
     let (resolve, world_id) = if let TypeConfig::Component(_) = project_cfg.project_type {
-        let component_bytes = tokio::fs::read(&artifact_path)
-            .await
-            .context("failed to read component bytes from built artifact path")?;
+        let component_bytes = tokio::fs::read(&artifact_path).await.with_context(|| {
+            format!(
+                "failed to read component bytes from built artifact path {}",
+                artifact_path.to_string_lossy()
```

Comment:
```suggestion
                "failed to read component bytes from built artifact path [{}]",
                artifact_path.display()
```

Can we not use `display()` here?

Reply:
I didn't know that existed! Neat